.github/workflows/dotnet.yml

```diff
@@ -34,34 +39,89 @@ jobs:
     - name: Package
       if: github.ref == 'refs/heads/main' && github.event_name == 'push'
       run: dotnet pack -c Release --no-restore --no-build --version-suffix "github$GITHUB_RUN_ID"
+
     - uses: actions/upload-artifact@v3
       if: github.ref == 'refs/heads/main' && github.event_name == 'push'
       with:
         name: nupkg
         path: nuget/*
 
     - name: Build Docs
-      if: github.ref == 'refs/heads/main' && github.event_name == 'push'
-      run: dotnet tool restore && cd ./docfx && dotnet docfx docfx_project/docfx.json
+      if: github.event_name == 'push'
+      run: |
+        dotnet tool restore
+        cd ./docfx
+        sed -i -E "s/%APP_VERSION%/${GITHUB_REF_NAME}/" docfx_project/docfx.json
+        dotnet docfx docfx_project/docfx.json
+
     - uses: actions/upload-artifact@v3
-      if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+      if: github.event_name == 'push'
       with:
         name: docfx-site
         path: docfx/docfx_project/_site/
 
   publish_docs:
-    # TODO: Publish two copies of the docs - one for tagged releases, and one for the latest commit
     name: Publish Documentation
     needs: build
-    if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+    if: github.event_name == 'push'
     runs-on: ubuntu-latest
 
+    env:
+      KEEP_TAG_VERSIONS: 5
+
     steps:
+      - name: Checkout current gh-pages
+        uses: actions/checkout@v3
+        with:
+          ref: gh-pages
+          path: site
+
+      - name: Clear version
+        working-directory: site
+        run: |
+          rm -rf ${GITHUB_REF_NAME}
+          echo "Cleared ${GITHUB_REF_NAME}"
+
       - name: Download documentation artifacts
         uses: actions/download-artifact@v3
         with:
           name: docfx-site
-          path: site
+          path: site/${{ github.ref_name }}
+
+      - name: Manage versions
+        shell: bash
+        working-directory: site
+        run: |
+          # Read versions
```

Comment:
This might be better suited as a separate script file.

Reply:
Any recommended place?
How about docfx/scripts or .github/scripts?
(Would have to rework the job to also checkout the branch in the `docs` job)

Reply:
docfx/scripts sounds good to me.